### PR TITLE
Add new VMware unified driver to supported list #2

### PR DESCRIPTION
**Pull request by frapposelli**
> This is a follow up from #2606 as the original fork was removed and unable to update the PR

This PR adds the new VMware unified driver to the supported list, the driver now supports Fusion and Workstation across Windows, Linux and macOS.

A deprecation note has been added for the bundled vmwarefusion driver.

Ref: #2118 #2526 machine-drivers/discussion#5 machine-drivers/docker-machine-driver-vmware#1

Fixes: machine-drivers/docker-machine-driver-vmware#1 #2118

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -176,6 +177,22 @@
   pruneopts = "NUT"
   revision = "3165313d6d3f973ec0b0ed3ec5a63b520e065d40"
 
+[[projects]]
+  digest = "1:63ede27834b468648817fb80cfb95d40abfc61341f89cb7a0d6779b6aa955425"
+  name = "github.com/google/go-github"
```
Comment (tstromberg) [inline Gopkg.lock]: Any idea what is pulling in this dependency? It seems unusual to include a github client into minikube.

Comment (frapposelli) [reply Gopkg.lock]: it looked weird to me as well, and I honestly don't know where it's coming from. 

Before submitting the PR I double checked and it is not coming from our `github.com/machine-drivers/docker-machine-driver-vmware` import, so I assumed that the `dep ensure` command pulled a dependency coming from somewhere else in the minikube codebase.

**Comment by mlebrun** [comment]
@frapposelli I updated Minikube to the 0.33.1 release to try out this driver and it seems like VMware Tools isn't installed? Which then makes Shared Folder support a bit unstable/unusable. I wasn't sure if there's anything I can do to get better Shared Folder support from Fusion (OSX) within Minikube? (I realize you're not full time Minikube maintainer, seeing if you have any input!) The `minikube --mount` method isn't that great when mounting a lot of files (or maybe just seems better suited to smaller projects?) so that's why I'm hoping for better Shared Folder support. Wasn't sure if you give any insight into that or not?

**Comment by mlebrun** [comment]
Seeing this which I just noticed when running verbose logs on `minikube start`:
```(minikube) DBG | executing: /Applications/VMware Fusion.app/Contents/Public/vmrun -gu docker -gp tcuser addSharedFolder /Users/{user}/.minikube/machines/minikube/minikube.vmx Users /Users
(minikube) DBG | executing: /Applications/VMware Fusion.app/Contents/Public/vmrun -gu docker -gp tcuser runScriptInGuest /Users/{user}/.minikube/machines/minikube/minikube.vmx /bin/sh [ ! -d /hosthome ]&& sudo mkdir /hosthome; sudo mount --bind /mnt/hgfs//hosthome /hosthome || [ -f /usr/local/bin/vmhgfs-fuse ]&& sudo /usr/local/bin/vmhgfs-fuse -o allow_other .host:/Users /hosthome || sudo mount -t vmhgfs -o uid=$(id -u),gid=$(id -g) .host:/Users /hosthome
(minikube) DBG | Guest program exited with non-zero exit code: 1```

**Pull request merged**